### PR TITLE
feat(grpc): endpoints for creating raw transaction

### LINKED ISSUE #351: GRPC Endpoints for creating raw transaction
We need GRPC Endpoints for creating raw Send, Bond, Unbond and Withdraw transactions. User should be able to set required parameters and get unsigned raw transaction.

### PR DESCRIPTION
**Pull request by kehiy**
## Related issue(s)

- Fixes #351

### CODE REVIEW

**Comment by kehiy** [comment]
@b00f can you have a review?

File: www/grpc/transaction_test.go [hunk 1]
```diff
@@ -105,3 +106,70 @@ func TestSendRawTransaction(t *testing.T) {
...
 	assert.Nil(t, conn.Close(), "Error closing connection")
 }
+
+func TestGetRawTransaction(t *testing.T) {
+	ts := testsuite.NewTestSuite(t)
+
+	conn, client := testTransactionClient(t)
+
+	t.Run("Transfer", func(t *testing.T) {
+		tx, privateKey := ts.GenerateTestTransferTx()
```
Comment (b00f) [inline www/grpc/transaction_test.go:116]: To prevent import shadowing, rename `tx` to `trx`

File: www/grpc/transaction.go [hunk 2]
```diff
@@ -86,6 +88,107 @@ func (s *transactionServer) CalculateFee(_ context.Context,
...
+		RawTransaction: rawTx,
+	}, nil
+}
+
+func (s *transactionServer) GetRawUnBondTransaction(_ context.Context,
+	req *pactus.GetRawUnBondTransactionRequest,
+) (*pactus.GetRawTransactionResponse, error) {
+	validatorAddr, err := crypto.AddressFromString(req.ValidatorAddress)
+	if err != nil {
+		return &pactus.GetRawTransactionResponse{}, err
```
Comment (b00f) [inline www/grpc/transaction.go:154]: Why not `nil, err`?

File: www/grpc/transaction.go [hunk 3]
```diff
@@ -86,6 +88,107 @@ func (s *transactionServer) CalculateFee(_ context.Context,
...
+			return &pactus.GetRawTransactionResponse{}, err
+		}
+	} else {
+		publicKey = nil
+	}
+
+	bondTx := tx.NewBondTx(req.LockTime, sender, receiver, publicKey, req.Stake, req.Fee, req.Memo)
+	rawTx, err := bondTx.Bytes()
+	if err != nil {
+		return &pactus.GetRawTransactionResponse{}, err
```
Comment (b00f) [inline www/grpc/transaction.go:141]: Why not `nil, err`?

File: www/grpc/transaction_test.go [hunk 4]
```diff
@@ -105,3 +106,70 @@ func TestSendRawTransaction(t *testing.T) {
...
+			PublicKey: "",
+			Fee:       tx.Fee(),
+			Memo:      tx.Memo(),
+		})
+		assert.Nil(t, err)
+		fmt.Println(res.RawTransaction)
+	})
+
+	t.Run("UnBond", func(t *testing.T) {
+		valAddr := ts.RandValAddress()
```
Comment (b00f) [inline www/grpc/transaction_test.go:147]: To be compatible with the rest of the test: use: `tx, privateKey := ts.GenerateTestUnbondTx()`

File: www/grpc/transaction_test.go [hunk 5]
```diff
@@ -105,3 +106,70 @@ func TestSendRawTransaction(t *testing.T) {
...
+		assert.Nil(t, err)
+		fmt.Println(res.RawTransaction)
+	})
+
+	t.Run("Bond", func(t *testing.T) {
+		tx, privateKey := ts.GenerateTestBondTx()
+
+		res, err := client.GetRawBondTransaction(tCtx, &pactus.GetRawBondTransactionRequest{
+			LockTime:  tx.LockTime(),
+			Sender:    privateKey.PublicKeyNative().AccountAddress().String(),
```
Comment (b00f) [inline www/grpc/transaction_test.go:135]: You can use `trx.Signer.Address()`

File: www/grpc/transaction_test.go [hunk 6]
```diff
@@ -105,3 +106,70 @@ func TestSendRawTransaction(t *testing.T) {
...
+		fmt.Println(res.RawTransaction)
+	})
+
+	t.Run("UnBond", func(t *testing.T) {
+		valAddr := ts.RandValAddress()
+		lockTime := ts.RandUint32(5000)
+
+		res, err := client.GetRawUnBondTransaction(tCtx, &pactus.GetRawUnBondTransactionRequest{
+			LockTime:         lockTime,
+			ValidatorAddress: valAddr.String(),
```
Comment (b00f) [inline www/grpc/transaction_test.go:152]: You can use `trx.Signer.Address()`

File: www/grpc/transaction.go [hunk 7]
```diff
@@ -86,6 +88,107 @@ func (s *transactionServer) CalculateFee(_ context.Context,
 	}, nil
 }
 
+func (s *transactionServer) GetRawTransferTransaction(_ context.Context,
+	req *pactus.GetRawTransferTransactionRequest,
+) (*pactus.GetRawTransactionResponse, error) {
+	sender, err := crypto.AddressFromString(req.Sender)
+	if err != nil {
+		return &pactus.GetRawTransactionResponse{}, err
```
Comment (b00f) [inline www/grpc/transaction.go:96]: Same here: `nil, err`

**Review by b00f — CHANGES_REQUESTED** [review]

**Comment by b00f** [comment]
@kehiy Everything looks good, Just small issues.

**Pull request closed without merging**